docs/topics/Permissions.md

```diff
@@ -179,17 +179,18 @@ Roles represent a set of permissions attached to a group of users. Roles have un
 
 ###### Role Structure
 
-| Field       | Type                                                                         | Description                                      |
-| ----------- | ---------------------------------------------------------------------------- | ------------------------------------------------ |
-| id          | snowflake                                                                    | role id                                          |
-| name        | string                                                                       | role name                                        |
-| color       | integer                                                                      | integer representation of hexadecimal color code |
-| hoist       | boolean                                                                      | if this role is pinned in the user listing       |
-| position    | integer                                                                      | position of this role                            |
-| permissions | string                                                                       | permission bit set                               |
-| managed     | boolean                                                                      | whether this role is managed by an integration   |
-| mentionable | boolean                                                                      | whether this role is mentionable                 |
-| tags?       | [role tags](#DOCS_TOPICS_PERMISSIONS/role-object-role-tags-structure) object | the tags this role has                           |
+| Field       | Type                                                                         | Description                                       |
+| ----------- | ---------------------------------------------------------------------------- | ------------------------------------------------- |
+| id          | snowflake                                                                    | role id                                           |
+| name        | string                                                                       | role name                                         |
+| color       | integer                                                                      | integer representation of hexadecimal color code  |
+| hoist       | boolean                                                                      | if this role is pinned in the user listing        |
+| icon        | ?string                                                                      | role [icon hash](#DOCS_REFERENCE/image-formatting)|
```

Comment:
I believe this can also be an emoji identifier/id, not sure which one

Reply:
Emojis are internally translated to a base64 image string in the discord client when you select an emoji.
It's always an string with an icon hash _as I saw_.
But I'll take another look into it.
Thanks!

Reply:
But then how would a bot set a role’s icon to a custom emoji? Would we have to upload the image again?

Reply:
Yes, as I mentioned to you a bit ago

Reply:
That's what I'm unsure about.
@Jupith Do you have an idea?

Reply:
https://github.com/Discord-Datamining/Discord-Datamining/commit/2f1451b4a03a1c0b0596173157084a7f45acdbca#commitcomment-56678733
https://github.com/Discord-Datamining/Discord-Datamining/commit/2f1451b4a03a1c0b0596173157084a7f45acdbca#commitcomment-56679016

Reply:
> But then how would a bot set a role’s icon to a custom emoji? Would we have to upload the image again?

yes, this is what the client does

Reply:
Emojis are uploaded as regular image files FWIW